lib/chef/provider/powershell_script.rb

```diff
@@ -54,7 +54,14 @@ def command
 
       def interpreter_path
         # Powershell.exe is always in "v1.0" folder (for backwards compatibility)
-        Chef::Util::PathHelper.join(basepath, "WindowsPowerShell", "v1.0", interpreter)
+        # pwsh is the other interpreter and we will assume that it is on the path.
+        # It will exist in different folders depending on the installed version.
+        # There can also be multiple versions installed. Depending on how it was installed,
+        # there might be a registry entry pointing to the installation path. The key will
+        # differ depending on version and architecture. It seems best to let the PATH
+        # determine the file path to use since that will provide the same pwsh.exe one
+        # would invoke from any shell.
+        interpreter == "powershell" ? Chef::Util::PathHelper.join(basepath, "WindowsPowerShell", "v1.0", "#{interpreter}.exe") : interpreter
```

Comment:
This is kind of a lot for a one-liner.

Reply:
haha. I'll make it multi line. I'm used to the rust linter that would have complained if it should be broken up.